### PR TITLE
Add Kullback-Leibler divergence

### PR DESCRIPTION
**Pull request by saromanov**
Hi!
I just implmeneted KL divergence, but at this moment without documentation, because i don't know which module is right for this. And in general, I don't know about need of of this feature for you :) Now, i put it in "probability". And created a separate file "entropy.js", because maybe someone will write another entropy related functions.  Now, entropy.js contains only ```kldivergence``` function.
Usage is trivial
```javascript
var mathjs = require('mathjs');
console.log(mathjs.kldivergence([0.5,0.4,0.2], mathjs.matrix([0.8,0.7,0.6]))); //0.3234281245654902
console.log(mathjs.kldivergence([0.5,0.4,0.2], [0.8,0.7,0.6])); //0.3234281245654902
```
You can put to the input javascript array or mathjs Matrix.

What do you think?

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks Sergey, your code looks neat! 

A few remarks:
- Can you rename the file `entropy.js` to `kldivergence.js`, corresponding to the name of the exported function?
- Looking at the code it looks like it should also work for BigNumbers and Fractions, but at the end you check whether the result is a `number` and if not return NaN. It would be great if you could add support for BigNumbers and fractions too. That would require to change the test `if (number.isNumber(result)) {` to something like `isNumeric(result)`, and adding unit tests for BigNumbers and Fractions.

**Comment by saromanov** [comment]
1. Maybe better, i will implement [classical entropy] (https://en.wikipedia.org/wiki/Entropy_(information_theory)) latter? Then bout entropy and kl divergence will be in the same file.
2. Ok, i'll do this. ```isNumeric(result)``` looks better.

Thank for the review!

**Comment by josdejong** [comment]
If you want to implement classical entropy too that would be nice of course, but please keep just one function per file.

**Comment by saromanov** [comment]
Sorry, but i'm not clear to understand about support BigNumbers and Fractions. I have read about these methods, and as i understand It is suitable for a methods that takes numbers as paremeters, but not arrays or matrices. 
I use number.isNumber(now, isNumeric), because construction
```javascript
sum(multiply(qnorm, log(dotDivide(qnorm, pnorm))));
````
returns ```{ [String: '0.8064758658669484 - 3.141592653589793i'] re: 0.8064758658669484, im: -3.141592653589793 }``` if one of input arrays contains negative numbers. But for the answer, complex numbers is not valid, so in this case, ```kldivergence``` returns NaN.

**Comment by josdejong** [comment]
> i'm not clear to understand about support BigNumbers and Fractions. 

`BigNumber` and `Fraction` are two data types which hold a numeric value, similar to `number`. You can read about it in the docs. So if you function works for numbers, it could also work for BigNumber and Fraction. It's not like a `Complex` number, which normally requires a completely different implementation.

With the changed check using `isNumeric`, I think that your current implementation just works for BigNumbers and fractions, so you can verify this with a few unit tests where you use BigNumbers and fractions instead of numbers in the arrays `q` and `p`.

File: test/function/probability/kldivergence.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        assert.equal(math.kldivergence(q, q), 0);
+        assert.equal(math.kldivergence(math.matrix(q), q), 0);
+        assert.equal(math.kldivergence(q, math.matrix(q)), 0);
+        assert.equal(math.kldivergence(math.matrix(q), math.matrix(q)), 0);
+    });
+
+    it('should return distance between two distrubutions', function(){
+        var q = [0.5,0.6,0.7];
+        var p = [0.4,0.5,0.6];
+        assert.notEqual(math.kldivergence(q, p), 0);
```
Comment (josdejong) [inline test/function/probability/kldivergence.test.js]: Is it enough to test whether the result is non-zero? Shouldn't you test what the value of the actual outcome is equal to? (Same at line 23)

**Comment by saromanov** [comment]
Well, I writed test for the ```BigNumber```and replace soft checks from the ```kldivergence```(notEqual) to the concrete results. Howewer, ```Fraction``` is still not supported, because when i tired to put ```Fraction``` objects to the array, i get exception. Probably the main cause in the function ```dotDivide```, but i'm not sure.

**Comment by josdejong** [comment]
Ah, `kldivergence` needs `log`, but `log` has no support for fractions, so indeed that will not work.

Thanks for adding the unit tests! I will merge your PR now.

**Pull request merged**